CMSgov/easi-app
# [EASI-3172] Upgrade go from 1.18 to 1.21

# EASI-3172

## Changes and Description

- https://github.com/CMSgov/mint-app/pull/682
- Modified references to go 1.18 to go 1.21
- Updated linters to use new versions that support go 1.21
- Updated docker volume mounts to only volume mount items that we need (comments explain at a high level why this is necessary, but tl;dr - mounting the `.git` directory in docker causes some issues with Go 1.21)

## How to test this change

- Start up the server locally with `scripts/dev up`, ensuring the app can build
- Use `scripts/dev db:seed` to seed the DB
- Play around with the app - give a smoke test to make sure things like submitting new requests works, System Profile works, login with Okta works, etc.
- Ensure E2E tests pass

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
